### ISSUE TITLE
MultiSelect (other dropdowns?) always have `aria-selected=false` even when selected

### ISSUE THREAD

**Issue by jhpedemonte** [carbon-design-system/carbon#6411]
## Detailed description

> What version of the Carbon Design System are you using?

10.15

> What did you expect to happen?

When selecting an option in `MultiSelect`, it has attribute `aria-selected="true"`.

> What happened instead?

The attribute `aria-selected` is always `false`, even when option is selected.

## Steps to reproduce the issue

1. Load storybook for `MultiSelect`: https://react.carbondesignsystem.com/?path=/story/multiselect--default
2. Open menu and inspect an option. Notice that `aria-selected` attribute is `false`.
3. Select an option, so that it has a checkmark.
4. Inspect that option. Notice that `aria-selected` is still `false` instead of `true`.

**Comment by joshblack** [comment]
cc @dakahn

**Comment by dakahn** [comment]
Downshift follows the [WAI-ARIA spec for listboxes](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html). This means that on 'highlight' -- either with your mouse or via a screen reader the `li` gets the `aria-selected="true"`attribute. We were breaking this functionality by setting this prop ourselves. PR #6440 fixes that bug and now it works to spec. '

Here's a link to an issue where the Downshift team addresses this behavior in case you're interested
https://github.com/downshift-js/downshift/issues/935#issuecomment-607958201

**Comment by jhpedemonte** [comment]
Interesting, I was confused: I thought `aria-selected=true` meant that the item was selected. But according to that spec, that's meant for an item that is "highlighted".

Should checked items in `MultiSelecte` have an attribute saying `checked=true`?

**Comment by jhpedemonte** [comment]
The spec (https://www.w3.org/TR/wai-aria-practices/examples/checkbox/checkbox-1/checkbox-1.html) specifies an `aria-checked` attribute. Also, shouldn't the list item have `role="checkbox"`?

**Comment by jhpedemonte** [comment]
Hi, any input on my last comment above ^^^?

**Comment by joshblack** [comment]
Hey @jhpedemonte! 👋  Sorry for the delay, I believe the current item is set with [`aria-activedescendant`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-activedescendant_attribute) if I remember correctly.
